### PR TITLE
Show errors when sign in with Facebook failed

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -82,6 +82,7 @@ def propagate_user_email_errors
     errors[:'emails.email'].each do |error|
       errors.add(:email, error)
     end
+    errors.delete(:'emails.email')
```
Comment (allenwq) [inline app/models/user.rb]: @lowjoel  I think this error should be deleted after propagate, otherwise it shows twice.

Comment (lowjoel) [inline app/models/user.rb]: simplify the function using Hash#delete. Do not access the hash so many times.

**Comment by lowjoel** [comment]
rebase please.

**Comment by allenwq** [comment]
@lowjoel  Done.

File: app/controllers/user/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -27,7 +27,9 @@ def sign_in_with_facebook
                         kind: t('user.omniauth_callbacks.facebook.kind')) if is_navigational_format?
     else
       session['devise.facebook_data'] = auth
-      redirect_to new_user_registration_path
+      redirect_to new_user_registration_path,
+                  danger: t('user.omniauth_callbacks.facebook.sign_in_failure',
+                            errors: @user.errors.full_messages.to_sentence)
```
Comment (lowjoel) [inline app/controllers/user/omniauth_callbacks_controller.rb]: final stickler: check you check all the failure strings -- do we use errors or error?

**Pull request merged**